src/coreclr/vm/bundle.cpp

```diff
@@ -80,7 +80,21 @@ BundleFileLocation Bundle::Probe(const SString& path, bool pathIsBundleRelative)
         }
     }
 
-    m_probe(utf8Path, &loc.Offset, &loc.Size);
+    // { fileSize, compressedSize }
+    INT64 sizes[2] = { 0, 0 };
+
+    m_probe(utf8Path, &loc.Offset, sizes);
+
+    if (sizes[1] != 0)
+    {
+        loc.Size = sizes[1];
+        loc.UncompresedSize = sizes[0];
```

Comment:
It just occurred to me - technically there's no reason to store the uncompressed size in the bundle headers. We would really need just a boolean. I assume it makes the code more efficient to know how much to allocate upfront to uncompress.

But what if the size in the header doesn't match the actual uncompressed size - we should have a test for that at the very least.

Reply:
Passing uncompressed allows to set up the storage for uncompressed data precisely, so it is slightly more convenient than a `bool`.

Reply:
If we have corrupted data we will not overrun the buffer, since we know the size. It does make sense to add a check and failfast if uncompressed size does not match, or compressed data was not all consumed. Good point!